addons/sale/tests/test_sale_order_discount.py

```diff
@@ -108,6 +108,41 @@ def test_percent_discount_above_100(self):
         with self.assertRaises(ValidationError):
             self.wizard.write({'discount_percentage': 1.1, 'discount_type': 'sol_discount'})
 
+    def test_discount_translation(self):
+        self.env['res.lang']._activate_lang('es_AR')
+        self.wizard.write({
+            'discount_percentage': 0.1,
+            'discount_type': 'so_discount',
+        })
+        self.wizard.sale_order_id.partner_id.lang = 'es_AR'
+        discount_product = self.wizard._get_discount_product()
+        self.wizard.action_apply_discount()
+        discount_line = self.sale_order.order_line[-1]
+        self.assertEqual(discount_line.name, "Descuento: 10.00%")
+
+    def test_discount_translation_tax_groups(self):
+        self.env['res.lang']._activate_lang('es_AR')
+        self.wizard.write({
+            'discount_percentage': 0.1,
+            'discount_type': 'so_discount',
+        })
+        self.wizard.sale_order_id.partner_id.lang = 'es_AR'
+        tax1 = self.env['account.tax'].create({
+            'name': 'test_translation1',
+            'amount_type': 'percent',
+            'amount': 10})
+        tax2 = self.env['account.tax'].create({
+            'name': 'test_translation2',
+            'amount_type': 'percent',
+            'amount': 20})
+        self.wizard.sale_order_id.order_line[0].tax_id = tax1
+        self.wizard.sale_order_id.order_line[1].tax_id = tax2
+        self.wizard.action_apply_discount()
+        discount_line_1 = self.sale_order.order_line[-1].name
+        discount_line_2 = self.sale_order.order_line[-2].name
+        self.assertEqual(discount_line_1, "Descuento: 10.00%- En los productos con los siguientes impuestos test_translation2")
+        self.assertEqual(discount_line_2, "Descuento: 10.00%- En los productos con los siguientes impuestos test_translation1")
```

Comment:
```suggestion
        self.assertEqual(
            self.sale_order.order_line[-2].name,
            "Descuento: 10.00%- En los productos con los siguientes impuestos 10% VAT",
        )
        self.assertEqual(
            self.sale_order.order_line[-1].name,
            "Descuento: 10.00%- En los productos con los siguientes impuestos 20% VAT",
        )
```
Sale styleguide requires lines to be 100 chars max